### PR TITLE
REF: remove freq arg from PeriodArray constructor

### PR DESCRIPTION
**Pull request by jbrockmendel**
Try to match the other constructors more closely

### CODE REVIEW

**Comment by mroeschke** [comment]
Does this need to be deprecated since it's somewhat public? https://pandas.pydata.org/docs/reference/api/pandas.arrays.PeriodArray.html

**Comment by jbrockmendel** [comment]
> Does this need to be deprecated since it's somewhat public?

OK, will update

**Comment by jbrockmendel** [comment]
updated as a deprecation + green

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -173,7 +173,8 @@ Deprecations
 - Deprecated the methods :meth:`Series.bool` and :meth:`DataFrame.bool` (:issue:`51749`)
 - Deprecated :meth:`DataFrame.swapaxes` and :meth:`Series.swapaxes`, use :meth:`DataFrame.transpose` or :meth:`Series.transpose` instead (:issue:`51946`)
 - Deprecated parameter ``convert_type`` in :meth:`Series.apply` (:issue:`52140`)
-
+- Deprecated ``freq`` parameter in :class:`PeriodArray` constructor, pass ``dtype`` instead (:issue:52462`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:176]:
```suggestion
- Deprecated ``freq`` parameter in :class:`PeriodArray` constructor, pass ``dtype`` instead (:issue:`52462`)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**